### PR TITLE
[Integration][Jira] Fix webhook search index lag causing missed issue creation (PORT-17308)

### PR DESCRIPTION
**Pull request by shariff-6**
# Description

What - Webhook processor now uses `POST /rest/api/3/search/jql` with `reconcileIssues` parameter instead of `GET` when verifying newly created/updated issues against JQL filters.

Why - Jira's search index is eventually consistent. When a `jira:issue_created` webhook arrives, the integration re-queries the Search API with the configured JQL filter, but ~20% of the time the issue isn't indexed yet. The issue doesn't pass the JQL filter and is incorrectly marked for deletion instead of creation. Issues only appear after the next resync (3min–8 hours). Zendesk #6713 / #5259.

How - Added `search_issues_by_ids()` method to `JiraClient` that uses `POST /rest/api/3/search/jql` with a JSON body containing `reconcileIssues` — a Jira API parameter that guarantees read-after-write consistency for specified issue IDs. The webhook processor now extracts `issue.id` from the payload and calls this method instead of `get_paginated_issues()`. The resync path (`get_paginated_issues` via GET) is completely untouched.

### Before (bug)

```mermaid
sequenceDiagram
    participant Jira
    participant Webhook as Webhook Processor
    participant Search as Jira Search API (GET)
    participant Port

    Jira->>Webhook: jira:issue_created {key: TEST-1, id: 12345}
    Webhook->>Search: GET /search/jql?jql=(project=TEST) AND key=TEST-1
    Note over Search: Search index hasn't<br/>caught up yet
    Search-->>Webhook: {issues: []} (empty — not indexed)
    Webhook->>Port: Mark TEST-1 for DELETION ❌
    Note over Port: Issue missing until<br/>next resync (3min–8hrs)
```

### After (fix)

```mermaid
sequenceDiagram
    participant Jira
    participant Webhook as Webhook Processor
    participant Search as Jira Search API (POST)
    participant Port

    Jira->>Webhook: jira:issue_created {key: TEST-1, id: 12345}
    Webhook->>Search: POST /search/jql {jql: "...", reconcileIssues: [12345]}
    Note over Search: reconcileIssues guarantees<br/>read-after-write consistency
    Search-->>Webhook: {issues: [{key: TEST-1, ...}]} ✅
    Webhook->>Port: Upsert TEST-1 ✅
    Note over Port: Issue appears immediately
```

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### Integration testing checklist

- [x] Integration able to create all default resources from scratch
- [x] Completed a full resync from a freshly installed integration and it completed successfully
- [x] Resync able to create entities
- [x] Resync able to update entities
- [x] Resync able to detect and delete entities
- [x] Resync finishes successfully
- [x] If new resource kind is added or updated in the integration, add example raw data, mapping and expected result to the `examples` folder in the integration directory.
- [x] If resource kind is updated, run the integration with the example data and check if the expected result is achieved
- [x] If new resource kind is added or updated, validate that live-events for that resource are working as expected
- [ ] Docs PR link [here](#)

### Preflight checklist

- [x] Handled rate limiting
- [x] Handled pagination
- [x] Implemented the code in async
- [x] Support Multi account

## File Walkthrough

| File | Change | Diff |
|------|--------|------|
| `integrations/jira/jira/client.py` | Add `search_issues_by_ids()` — POST-based search with `reconcileIssues` for read-after-write consistency | +19/-0 |
| `integrations/jira/webhook_processors/issue_webhook_processor.py` | Extract `issue.id` from payload, call `search_issues_by_ids` instead of `get_paginated_issues` | +9/-10 |
| `integrations/jira/tests/test_client.py` | Add 2 tests for `search_issues_by_ids` (success + empty response) | +50/-0 |
| `integrations/jira/tests/test_processors.py` | Update all issue webhook tests to use `search_issues_by_ids` with `AsyncMock`; fix `async def` on JQL parentheses test | +38/-69 |

## API Documentation

- [POST /rest/api/3/search/jql - reconcileIssues](https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-issue-search/#api-rest-api-3-search-jql-post)

### CODE REVIEW

File: integrations/jira/webhook_processors/issue_webhook_processor.py [hunk 1]
```diff
@@ -28,6 +28,7 @@ async def handle_event(
         client = create_jira_client()
         config = cast(JiraIssueConfig, resource_config)
         issue_key = payload["issue"]["key"]
+        issue_id = int(payload["issue"]["id"])
```
Comment (emekanwaoma) [inline integrations/jira/webhook_processors/issue_webhook_processor.py:31]: let's validate these in validate_payload method

File: integrations/jira/jira/client.py [hunk 2]
```diff
@@ -307,6 +307,25 @@ async def get_paginated_issues(
         ):
             yield issues
 
+    async def search_issues_by_ids(
+        self,
+        jql: str,
+        issue_ids: list[int],
+        fields: str = "*all",
```
Comment (emekanwaoma) [inline integrations/jira/jira/client.py]:
```suggestion
        fields: Optional[str] = None,
```

File: integrations/jira/jira/client.py [hunk 3]
```diff
@@ -307,6 +307,25 @@ async def get_paginated_issues(
...
+    ) -> list[dict[str, Any]]:
+        """Filter known issues against a JQL query with read-after-write consistency"""
+
+        url = f"{self.api_url}/search/jql"
+        fields_list = fields.split(",") if fields != "*all" else ["*all"]
```
Comment (emekanwaoma) [inline integrations/jira/jira/client.py]:
```suggestion
        fields_list = (
            ["*all"]
            if not fields or fields == "*all"
            else [f.strip() for f in fields.split(",") if f.strip()]
        )

```

**Review by emekanwaoma — CHANGES_REQUESTED** [review]
Left comments

I see we didn't include the retry. Any reason?

File: integrations/jira/webhook_processors/issue_webhook_processor.py [hunk 4]
```diff
@@ -72,4 +71,4 @@ async def authenticate(self, payload: EventPayload, headers: EventHeaders) -> bo
         return True
 
     async def validate_payload(self, payload: EventPayload) -> bool:
-        return True
+        return "webhookEvent" in payload and "issue" in payload
```
Comment (emekanwaoma) [inline integrations/jira/webhook_processors/issue_webhook_processor.py]: what we need to validate here is that
- the key "issue" exists in payload
- the key "issue['key']" exists in payload
- the key "issue['id']" exists in payload

Comment (shariff-6) [reply integrations/jira/webhook_processors/issue_webhook_processor.py]: Why not let it fail ? Wont this mask an issue from jira

File: integrations/jira/jira/client.py [hunk 5]
```diff
@@ -307,6 +307,32 @@ async def get_paginated_issues(
         ):
             yield issues
 
+    async def search_issues_by_ids(
```
Comment (matan84) [inline integrations/jira/jira/client.py]: Rename this name to refernce the reconcileIssues / change the search issues to get optional arguments which will perform this instead of writing a new function

**Review by matan84 — CHANGES_REQUESTED** [review]
left commnet

File: integrations/jira/jira/client.py (hunk 5 above)
Comment (shariff-6) [reply integrations/jira/jira/client.py]: I went with the rename, changing the search requires migrating the resync to use the new search endpoint which we wanted to plan for.
